osc-server/src/test/java/org/osc/core/broker/service/servicefunctionchain/AddServiceFunctionChainServiceTest.java

```diff
@@ -0,0 +1,169 @@
+    
+    private VirtualizationConnector vc;
+    
+    @Before
+    public void testInitialize() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        // @InjectMocks should do this
+        this.service.validator = this.validatorMock;
+
+        this.em = InMemDB.getEntityManagerFactory().createEntityManager();
+
+        this.txControl.setEntityManager(this.em);
+
+        Mockito.when(this.dbMgr.getTransactionalEntityManager()).thenReturn(this.em);
+        Mockito.when(this.dbMgr.getTransactionControl()).thenReturn(this.txControl);
+
+
+        populateDatabase();
+
+        this.request = new AddOrUpdateServiceFunctionChainRequest();
+        this.sfcDto = new ServiceFunctionChainDto();
+        this.sfcDto.setName("sfcName");
+        this.request.setVcId(this.vc.getId());
+        this.request.setDto(this.sfcDto);
+    }
+    
+    @After
+    public void testTearDown() {
+        InMemDB.shutdown();
+    }
+    
+    private void populateDatabase() {
+        this.em.getTransaction().begin();
+
+        this.vc = new VirtualizationConnector();
+        this.vc.setVirtualizationType(VirtualizationType.OPENSTACK);
+        this.vc.setVirtualizationSoftwareVersion("vcSoftwareVersion");
+        this.vc.setName("vcName");
+        this.vc.setProviderIpAddress("127.0.0.1");
+        this.vc.setProviderUsername("User-name");
+        this.vc.setProviderPassword("********");
+        this.vc.setId(1L);
+
+
+        this.em.persist(this.vc);
+        this.em.getTransaction().commit();
+     }
+
+
+    @Test
+    public void testDispatch_WithNullRequest_ThrowsNullPointerException() throws Exception {
+        
+        // Arrange.
+        this.exception.expect(NullPointerException.class);
+
+        // Act.
+        this.service.dispatch(null);
+    }
+    
+    @Test
+    public void testDispatch_WithInvalidVcId_ThrowsVmidcBrokerValidationException() throws Exception {
+        
+        // Arrange.
+    	this.request.setVcId(2L);
+        this.exception.expect(VmidcBrokerValidationException.class);
+
+        // Act.
+        this.service.dispatch(this.request);
+    }
+    
+    @Test
+    public void testDispatch_WithNullVcId_ThrowsIllegalArgumentException() throws Exception {
+        
+        // Arrange.
+    	this.request.setVcId(null);
+        this.exception.expect(IllegalArgumentException.class);
+
+        // Act.
+        this.service.dispatch(this.request);
+    }
+      
+    @Test
+    public void testDispatch_WhenAddRequestIsValid_ValidationSucceeds() throws Exception {
+    	 // Arrange.
+    	//AddorUpdateServiceFunctionChainRequest testRequest = new AddorUpdateServiceFunctionChainRequest();
+    	//testRequest = this.request;
+
+        // Act.
+    	BaseJobResponse response = this.service.dispatch(this.request);
+    	// Assert.
+        Mockito.verify(this.validatorMock).validateForCreate(this.request.getDto());
+    	Assert.assertNotNull("The returned response should not be null.", response);
```

Comment:
Also ensure to check whether the id in the response is correct. See my latest PR for an example